JitsiMeetJS.js

```diff
@@ -246,11 +254,14 @@ export default {
      *
      * @param {boolean} (firePermissionPromptIsShownEvent) - if event
      *      JitsiMediaDevicesEvents.PERMISSION_PROMPT_IS_SHOWN should be fired
+     * @param originalOptions - internal use only, to be able to store the
+     * originally requested options.
      * @returns {Promise.<{Array.<JitsiTrack>}, JitsiConferenceError>}
      *     A promise that returns an array of created JitsiTracks if resolved,
      *     or a JitsiConferenceError if rejected.
      */
-    createLocalTracks(options = {}, firePermissionPromptIsShownEvent) {
+    createLocalTracks(
+            options = {}, firePermissionPromptIsShownEvent, originalOptions) {
```

Comment:
Why is it necessary to have "originalOptions"? For example, are the mutations of options so deep that keeping a shallow copy of the original is insufficient?

Reply:
Yep, `createLocalTracks ` is calling itself and so the original copy is lost.